### ISSUE TITLE
Fennec migration: Fennec remains in the Recent apps after migration

### ISSUE THREAD

**Issue by ebalazs-sv** [mozilla-mobile/android-components#5194]
## Preconditions:
Have Fennec Release installed.
## Steps to reproduce
1. Replace Fennec with Fenix using a build from https://github.com/mozilla-mobile/fenix/wiki/Fennec-Migration;
2. Open Fenix;
3. Tap on the Recent apps device button;
4. Observe the Recent apps list.
### Expected behavior
Fennec is not displayed in Recent apps.
### Actual behavior
Fennec is available in Recent apps and its not usable.
### Notes:
[Video
](https://drive.google.com/file/d/1q8RA0fgw5LineltISbFnevTDKV-nLPFF/view?usp=sharing)
Not reproducible on Samsung Galaxy S8 (Android 9).
### Device information

* Android device: Pixel 2 (Android 9)
* Fenix version: Nightly 191102 GV 71

**Comment by pocmo** [comment]
Wow, weird. I wonder if that is a launcher bug since the app usually gets killed on update (and then may drop out of the recent apps list?). I can try to debug this a bit on my devices and see if that happens too - I do not have a Samsung phone though.

**Comment by ebalazs-sv** [comment]
Adding an unfiltered device log: 
[recentapps.txt](https://github.com/mozilla-mobile/android-components/files/3904724/recentapps.txt)

**Comment by pocmo** [comment]
In the log I see multiple entries like this AFTER the migration seems to have already happened (at least I see logs from Fenix code; I hope that's not another Fenix installation):

> 2019-11-29 12:14:23.130 1174-4212/? I/ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10100000 pkg=org.mozilla.firefox cmp=org.mozilla.firefox/org.mozilla.gecko.BrowserApp} from uid 10427

In this case the launcher is trying to re-launch the app with the intent it was started initially. This intent handler does not exist anymore after the upgrade.

We should try those STR again after fixing https://github.com/mozilla-mobile/fenix/issues/4976 - In that issue we will try to restore the previous intent filter to keep the launcher icons on the home screen. This should also fix this issue here if my assumption above is correct.

**Comment by ebalazs-sv** [comment]
@pocmo This is still reproducible after mozilla-mobile/fenix#4976 was fixed and verified.
Also reproducible for Beta and Nightly migration builds with Pixel 2 (Android 9), Nexus 6P (Android 8.1.0) if Fennec is in Recent apps when migrating.

**Comment by sv-ohorvath** [comment]
Still reproducible on the 1/16 build.

**Comment by ebalazs-sv** [comment]
This is reproducible on Fennec Nightly migration build 2, 01/17. The behavior is different now: Fennec can be tapped and Fenix is displayed.
[Video](https://drive.google.com/file/d/1IwUZkTUK6PWQPGuTFC4VA0NJT6LIiid8/view?usp=sharing)

**Comment by pocmo** [comment]
I tried to reproduce this on an emulator. Fennec was still in recent apps, but clicking it launched back into Fenix. Will do some more tests.

**Comment by pocmo** [comment]
> The behavior is different now: Fennec can be tapped and Fenix is displayed.
Video

Ah, I see you already posted that.

This may be the best we can do here. I'll look at this again.

**Comment by vesta0** [comment]
I watched the video and I think we should fix this if we can, it would look confusing to users.

**Comment by pocmo** [comment]
Looking at app  tasks during the migration:

```
                  SKDBG  W  Found 2 tasks
                         W  Task 593
                         W   - ComponentInfo{org.mozilla.fennec_aurora/org.mozilla.fennec_aurora.App}
                         W   - Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.mozilla.fennec_aurora/org.mo
                            zilla.fenix.MigrationDecisionActivity (has extras) }
                         W  Task -1
                         W   - null
                         W   - Intent { act=android.intent.action.MA
```

**Comment by pocmo** [comment]
The only idea I can come up with is to look through the running app tasks, try to identify an old fennec task and then try to get rid of it. I'm not sure how successful that will be and whether that may introduce more problems (like us removing our own task that shows the migration UI). 🤔

**Comment by pocmo** [comment]
Should be fixed with the next build.

**Comment by ebalazs-sv** [comment]
I tested this on Beta Migration build 02/19 with Samsung Galaxy Note 8 (Android 9), Huawei P9 Lite (Android 7) and Fennec does not remain in the Recent apps after migration.
I will leave this on Ready for QA until it will also be verified on the official Beta migration build.

**Comment by LaurentiuApahideanSV** [comment]
Verified as fixed on Beta Migration Build #1, Firefox Preview Beta 74.0.0-beta.1 (Build #2015724883) using a Samsung Galaxy S5 (Android 6.0.1) and OnePlus 6T (Android 9). Firefox is no longer present in the recent app list after a migration is performed.
